clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteredResourceConfigurationParser.java

```diff
@@ -104,11 +128,47 @@ public ResourcePool parseResourceConfiguration(final Element fragment) {
 
   @Override
   public Element unparseResourcePool(ResourcePool resourcePool) {
-    throw new UnsupportedOperationException("Not yet implemented");
+    return unparseConfig(resourcePool);
+  }
+
+  @Override
+  public Element createRootElement(Document doc, ResourcePool resourcePool)
+  {
+    Element rootElement = doc.createElementNS(EHCACHE_NAMESPACE, EHCACHE_NAMESPACE_PREFIX + COLON + RESOURCE_ELEMENT_NAME);
+    rootElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + EHCACHE_NAMESPACE_PREFIX, EHCACHE_NAMESPACE);
+    if (ClusteredResourcePoolImpl.class.isInstance(resourcePool)) {
+      Element clusterElement = doc.createElementNS(NAMESPACE.toString(), RESOURCE_NAMESPACE_PREFIX + COLON + CLUSTERED_ELEMENT_NAME);
+      clusterElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + RESOURCE_NAMESPACE_PREFIX, NAMESPACE
+        .toString());
+      rootElement.appendChild(clusterElement);
+    } else if (DedicatedClusteredResourcePoolImpl.class.isInstance(resourcePool)) {
+      DedicatedClusteredResourcePoolImpl dedicatedClusteredResourcePool = (DedicatedClusteredResourcePoolImpl) resourcePool;
+      Element dedicatedClusterElement = doc.createElementNS(NAMESPACE.toString(), RESOURCE_NAMESPACE_PREFIX + COLON + DEDICATED_ELEMENT_NAME);
+      dedicatedClusterElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + RESOURCE_NAMESPACE_PREFIX, NAMESPACE
+        .toString());
+      if (dedicatedClusteredResourcePool.getFromResource() != null) {
+        dedicatedClusterElement.setAttribute(FROM_ELEMENT_NAME, dedicatedClusteredResourcePool.getFromResource());
+      }
+      dedicatedClusterElement.setAttribute(UNIT_ELEMENT_NAME, dedicatedClusteredResourcePool.getUnit().toString());
+      dedicatedClusterElement.setTextContent(String.valueOf(dedicatedClusteredResourcePool.getSize()));
+      rootElement.appendChild(dedicatedClusterElement);
+    } else if (SharedClusteredResourcePoolImpl.class.isInstance(resourcePool)) {
+      SharedClusteredResourcePoolImpl sharedClusteredResourcePool = (SharedClusteredResourcePoolImpl) resourcePool;
+      Element sharedClusterElement = doc.createElementNS(NAMESPACE.toString(), RESOURCE_NAMESPACE_PREFIX + COLON + SHARED_ELEMENT_NAME);
+      sharedClusterElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + RESOURCE_NAMESPACE_PREFIX, NAMESPACE
+        .toString());
+      sharedClusterElement.setAttribute(SHARING_ELEMENT_NAME, sharedClusteredResourcePool.getSharedResourcePool());
+      rootElement.appendChild(sharedClusterElement);
+    }
+    return rootElement;
   }
 
   @Override
   public Set<ResourceType<?>> getResourceTypes() {
-    return null;
+    Set<ResourceType<?>> resourceType = new HashSet<>(3);
```

Comment:
I think you can do`Collections.unmodifiableSet(new HashSet<>(Arrays.asList(ClusteredResourceType.values()));`. Maybe this code should move to `ClusteredResourceType`